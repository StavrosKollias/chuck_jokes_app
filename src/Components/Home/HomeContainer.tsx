import React, { useState, useEffect } from "react";
import { IJoke } from "../../Interfaces/IJokes";
import { getMultipleRandomJokes } from "../../utils/apis";
import CardsContainer from "../Cards/CardsContainer/CardsContainer";
import WelcomeComponent from "../WelcomeCard/WelcomeComponent";
import { IHomeContainerProps } from "./IHomeContainerProps";
import "./HomeContainer.scss";

const INITIAL_STATE: Array<IJoke> = null;

const HomeContainer: React.FC<IHomeContainerProps> = (props) => {
   props.onRender("");
   const [state, setState] = useState<Array<IJoke>>(INITIAL_STATE);
   const getFourRandomJokes = async (numberJokes: number) => {
      const arrayJokes = await getMultipleRandomJokes(numberJokes);
      if (arrayJokes) return arrayJokes;
   };

   useEffect(() => {
      if (!state) {
         getFourRandomJokes(4).then((data) => {
            setState(data);
         });
      }
   });

   useEffect(() => {
      return () => {};
   });

   return (
      <div className="container py-1 home-container">
         <WelcomeComponent />
         {state && <CardsContainer jokes={state} />}
      </div>
   );
};

export default HomeContainer;
