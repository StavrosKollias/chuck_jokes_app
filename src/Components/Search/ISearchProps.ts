export interface ISearchProps {
   onRender?(e: any): void;
}
