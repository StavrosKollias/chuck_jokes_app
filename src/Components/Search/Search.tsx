import React from "react";
import JockesContainer from "../Jokes/JokesContainer/JokesContainer";

interface ISearchProps {
   onRender?(e: any): void;
}
const Search: React.FC<ISearchProps> = (props) => {
   return (
      <section className="search-container">
         <JockesContainer multiple={true} filter={true} onRender={(e) => props.onRender(e)} />
      </section>
   );
};

export default Search;
