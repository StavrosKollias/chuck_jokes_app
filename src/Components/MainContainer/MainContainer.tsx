import React from "react";
import { Switch, Route, BrowserRouter } from "react-router-dom";
import Footer from "../Footer/Foooter";
import Header from "../Header/Header";
import HomeContainer from "../Home/HomeContainer";
import List from "../List/List";
import Random from "../Random/Random";
import Search from "../Search/Search";

const MainContainer: React.FC = () => {
   const loadFunctionComponents = (e: any) => {
      return window.location.pathname;
   };

   return (
      <main className="wrapper">
         <BrowserRouter>
            <Header />
            <div className="container">
               <Switch>
                  <Route path="/" exact match render={() => <HomeContainer onRender={(e) => loadFunctionComponents(e)} />} />
                  <Route path="/random-joke" exact render={() => <Random onRender={(e) => loadFunctionComponents(e)} />} />
                  <Route path="/joke-list" exact render={() => <List onRender={(e) => loadFunctionComponents(e)} />} />
                  <Route path="/search" exact render={() => <Search onRender={(e) => loadFunctionComponents(e)} />} />
               </Switch>
            </div>
         </BrowserRouter>
         <Footer />
      </main>
   );
};

export default MainContainer;
