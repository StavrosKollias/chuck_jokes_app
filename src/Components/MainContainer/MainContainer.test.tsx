import React from "react";
import { shallow } from "enzyme";

import MainContainer from "./MainContainer";
import HomeContainer from "../Home/HomeContainer";
import Search from "../Search/Search";
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import { render } from "@testing-library/react";

describe("MainContainer behaviour", () => {
   let component;
   beforeAll(() => {
      component = shallow(<MainContainer />);
   });
   test("Renders the component", () => {
      expect(component).toHaveLength(1);
      expect(component).toMatchSnapshot();
   });
});
