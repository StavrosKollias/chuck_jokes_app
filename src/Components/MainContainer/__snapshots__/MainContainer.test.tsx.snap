// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContainer behaviour Renders the component 1`] = `
<main
  className="wrapper"
>
  <Component>
    <Header />
    <div
      className="container"
    >
      <Switch>
        <Route
          exact={true}
          match={true}
          path="/"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/random-joke"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/joke-list"
          render={[Function]}
        />
        <Route
          exact={true}
          path="/search"
          render={[Function]}
        />
      </Switch>
    </div>
  </Component>
  <Footer />
</main>
`;
